src/main/java/com/openlattice/datastore/search/controllers/SearchController.java

```diff
@@ -555,75 +533,124 @@ public SearchResult executeFQNPropertyTypeSearch( @RequestBody FQNSearchTerm sea
+            if ( !groupedEntitySets.get( false ).isEmpty() ) {
+                var regularEntitySetIds = groupedEntitySets.get( false ).stream()
+                        .map( Map.Entry::getKey ).collect( Collectors.toSet() );
+                result = searchGraphService.executeEntityNeighborIdsSearch(
+                        new EntityNeighborsFilterBulk(
+                                filter.getEntityKeyIds().entrySet().stream()
+                                        .filter( entry -> regularEntitySetIds.contains( entry.getKey() ) )
+                                        .collect( Collectors.toMap( Map.Entry::getKey, Map.Entry::getValue ) ),
+                                filter.getSrcEntitySetIds(),
+                                filter.getDstEntitySetIds(),
+                                filter.getAssociationEntitySetIds()
+                        ),
+                        principals );
+            } else if ( !groupedEntitySets.get( true ).isEmpty() ) {
+                var linkedEntitySetIds = groupedEntitySets.get( true ).stream()
+                        .map( Map.Entry::getKey ).collect( Collectors.toSet() );
+                result.putAll(
+                        searchGraphService.executeLinkingEntityNeighborIdsSearch(
+                                groupedEntitySets.get( true ).stream()
+                                        .collect( Collectors.toMap( Map.Entry::getKey, Map.Entry::getValue ) ),
+                                new EntityNeighborsFilterBulk(
+                                        filter.getEntityKeyIds().entrySet().stream()
+                                                .filter( entry -> linkedEntitySetIds.contains( entry.getKey() ) )
+                                                .collect( Collectors.toMap( Map.Entry::getKey, Map.Entry::getValue ) ),
+                                        filter.getSrcEntitySetIds(),
+                                        filter.getDstEntitySetIds(),
+                                        filter.getAssociationEntitySetIds()
+                                ),
+                                principals ) );
             }
         }
 
 
         /* audit */
+        int expectedSize = 0;
+        if ( !result.isEmpty() ) {
+            expectedSize = result.size();
 
-        SetMultimap<UUID, UUID> neighborsByEntitySet = HashMultimap.create();
+            var associationMap = result.values().iterator().next();
+            if ( !associationMap.isEmpty() ) {
+                expectedSize *= associationMap.size();
+
+                var associations = associationMap.values().iterator().next();
+                if ( !associations.isEmpty() ) {
+                    expectedSize *= associations.size();
+
+                    if ( !associations.values().iterator().next().isEmpty() ) {
+                        expectedSize *= associations.values().iterator().next().size();
+                    }
+                }
+            }
+        }
```

Comment:
Okay so in this case I was just hoping to get a heuristic value for the map size. Generally, calculating the map size by iterating is way way way less performant than just leaving the map un-sized so we should probably just revert to un-sized here

Reply:
👍